WordPress/Classes/ViewRelated/Comments/CommentContentTableViewCell.swift

```diff
@@ -0,0 +1,221 @@
+import UIKit
+
+class CommentContentTableViewCell: UITableViewCell, NibReusable {
+
+    // determines the state of the like button.
+    enum LikeState {
+        case unliked
+        case liked
+    }
+
+    // all the available images to display for the accessory button.
+    enum AccessoryButtonType {
+        case share
+        case threeDots
+    }
+
+    // MARK: - Public Properties
+
+    var displayName: String? = "" {
+        didSet {
+            nameLabel?.setText(displayName ?? "")
+        }
+    }
+
+    var displayDate: String? = "" {
+        didSet {
+            dateLabel?.setText(displayDate ?? "")
+        }
+    }
+
+    var accessoryButtonType: AccessoryButtonType = .share {
+        didSet {
+            accessoryButton.setImage(accessoryButtonImage, for: .normal)
+        }
+    }
+
+    var numberOfLikes: Int = 0 {
+        didSet {
+            updateLikeButton()
+        }
+    }
+
+    var likeButtonState: LikeState = .unliked {
+        didSet {
+            updateLikeButton()
+        }
+    }
+
+    // MARK: Button Tap Handlers
+
+    var nameLabelTapAction: (() -> Void)? = nil
+
+    var accessoryButtonAction: (() -> Void)? = nil
+
+    var replyButtonAction: (() -> Void)? = nil
+
+    var likeButtonAction: (() -> Void)? = nil
+
+    // MARK: Component Visibility
+
+    // Hides the accessory button if the value is false.
+    var accessoryButtonEnabled: Bool = true {
```

Comment:
All these button states can be private ya?

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/17109/commits/05d5a31e08a13244e2a493f9cf2aea0b40f07f88!